src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunction.cs

```diff
@@ -1,20 +1,95 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Text.Json;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Shared.Collections;
+using Microsoft.Shared.Diagnostics;
 
 namespace Microsoft.Extensions.AI;
 
 /// <summary>Represents a function that can be described to an AI service and invoked.</summary>
 [DebuggerDisplay("{DebuggerDisplay,nq}")]
 public abstract class AIFunction : AITool
 {
-    /// <summary>Gets metadata describing the function.</summary>
-    public abstract AIFunctionMetadata Metadata { get; }
+    /// <summary>The description of the function.</summary>
+    private readonly string _description = string.Empty;
+
+    /// <summary>The JSON schema describing the function and its input parameters.</summary>
+    private readonly JsonElement _jsonSchema = AIJsonUtilities.DefaultJsonSchema;
+
+    /// <summary>Optional additional properties in addition to the named properties already available on this class.</summary>
+    private readonly IReadOnlyDictionary<string, object?> _additionalProperties = EmptyReadOnlyDictionary<string, object?>.Instance;
+
+    /// <summary>Gets the name of the function.</summary>
+    public abstract string Name { get; }
+
+    /// <summary>Gets a description of the function, suitable for use in describing the purpose to a model.</summary>
+    [AllowNull]
+    public virtual string Description
+    {
+        get => _description;
+        init => _description = value ?? string.Empty;
+    }
+
+    /// <summary>
+    /// Gets the underlying <see cref="MethodInfo"/> that this <see cref="AIFunction"/> might be wrapping.
+    /// </summary>
+    /// <remarks>
+    /// Provides additional metadata on the function and its signature. Implementations not wrapping .NET methods may return <see langword="null"/>.
+    /// </remarks>
+    public virtual MethodInfo? UnderlyingMethod { get; init; }
```

Comment:
Same here. I guess it's a question for everything here... how are we deciding some things are init vs abstract?

Reply:
I reserved abstract for the properties that are required. Everything else I left as virtual to make overriding more convenient but they don't need to be.

Reply:
Can we also remove the "init"?